tests/ProgramTestTests/UserInput/ClickLinkTest.elm

```diff
@@ -75,6 +76,16 @@ all =
                     |> ProgramTest.start ()
                     |> ProgramTest.clickLink "SPA" "#search"
                     |> ProgramTest.expectModel (Expect.equal "<INIT:/>;GoToSearch")
+        , test "internal (single-page app) link causes url change" <|
+            \() ->
+                TestingProgram.application SimulatedEffect.None
+                    |> ProgramTest.clickLink "SPA" "/search?q=query"
+                    |> ProgramTest.expectModel (Expect.equal [ "OnUrlChange: https://example.com/search?q=query" ])
```

Comment:
`ProgramTest.ensureBrowserUrl` and/or `ensureBrowserHistory` might be appropriate here too.

Reply:
Done